tests/meta-mender-toradex-nxp-ci/conf/machine/colibri-imx7-mender.conf

```diff
@@ -41,6 +41,12 @@ IMAGE_ROOTFS_MAXSIZE = "307200"
 # meta-mender-core/recipes-bsp/u-boot/u-boot-fw-utils-mender.inc we need ship
 # our own config instead.
 MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET = "0x380000"
+# board configuration is set up with CONFIG_ENV_RANGE = 2 * CONFIG_ENV_SIZE,
+# hence we need to override the offset of redundant environment otherwise the
+# default value (as calculated by u-boot-mender-common.inc) would end up inside
+# the backup space of the main environment
+MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_1 = "${MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET}"
+MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_2 = "0x3c0000"
```

Comment:
I think this part should be unnecessary if we take `CONFIG_ENV_RANGE` into account in our calculations, as per my other comment.

Reply:
Basically, what you're suggesting is:
* add MENDER_BOOTENV_RANGE, defaults to unset
* update the integration patch to define CONFIG_ENV_RANGE iff MENDER_BOOTENV_RANGE is set
* verify that MENDER_BOOTENV_RANGE is a multiple of MENDER_BOOTENV_SIZE
* update `meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc` to account for MENDER_BOOTENV_RANGE when calculating the offset of redundant environment
* add check that the environments do not overlap

is that right? Looks doable.

OTOH, I would still keep the weak assignment to `MENDER_UBOOT_ENV_STORAGE_OFFSET_{1,2}` (+ a check that environments don't overlap). Should the user decide to override the offsets manually they surely knows better where they want the environment to be placed. Mender integration code should probably not interfere with this.

Reply:
> is that right? Looks doable.

Yes, exactly!

> OTOH, I would still keep the weak assignment to MENDER_UBOOT_ENV_STORAGE_OFFSET_{1,2} (+ a check that environments don't overlap). Should the user decide to override the offsets manually they surely knows better where they want the environment to be placed. Mender integration code should probably not interfere with this.

Yes, absolutely. Make the automagic work 80% of the time, and then make it easily overridable when it's being stupid! :-)